moveit_plugins/moveit_fake_controller_manager/src/moveit_fake_controller_manager.cpp

```diff
@@ -35,152 +35,156 @@
+        {
+          // @note: would put this in InterpolatingController's constructor
+          // but they disabled WallRate::operator= for some reason and that complicates things
+          double rate = 10.0;
+          std::string fake_interp_rate_param =
+              PARAM_BASE_NAME + "." + controller_name + ".fake_interpolating_controller_rate";
+          if (node_->has_parameter(fake_interp_rate_param))
+            node_->get_parameter(fake_interp_rate_param, rate);
+          controllers_[controller_name].reset(
+              new InterpolatingController(controller_name, controller_joints, pub_, rate));
+        }
         else
-          ROS_ERROR_STREAM("Unknown fake controller type: " << type);
+          RCLCPP_ERROR_STREAM(LOGGER, "Unknown fake controller type: " << type);
 
         moveit_controller_manager::MoveItControllerManager::ControllerState state;
-        state.default_ = controller_list[i].hasMember("default") ? (bool)controller_list[i]["default"] : false;
         state.active_ = true;
 
-        controller_states_[name] = state;
+        std::string default_state_param = PARAM_BASE_NAME + "." + controller_name + "default";
+        if (node_->has_parameter(default_state_param))
+          node_->get_parameter(default_state_param, state.default_);
+        controller_states_[controller_name] = state;
       }
       catch (...)
       {
-        ROS_ERROR_NAMED("MoveItFakeControllerManager", "Caught unknown exception while parsing controller information");
+        RCLCPP_ERROR(LOGGER, "Caught unknown exception while parsing controller information");
       }
     }
   }
 
-  sensor_msgs::JointState loadInitialJointValues(XmlRpc::XmlRpcValue& param) const
+  // TODO: codebase wide refactoring for XmlRpc
+  sensor_msgs::msg::JointState loadInitialJointValues(const std::string& param_base_name) const
   {
-    sensor_msgs::JointState js;
+    sensor_msgs::msg::JointState js;
 
-    if (param.getType() != XmlRpc::XmlRpcValue::TypeArray || param.size() == 0)
-    {
-      ROS_ERROR_ONCE_NAMED("loadInitialJointValues", "Parameter 'initial' should be an array of (group, pose) "
-                                                     "structs.");
-      return js;
-    }
-
-    robot_model_loader::RobotModelLoader robot_model_loader(ROBOT_DESCRIPTION);
-    const moveit::core::RobotModelPtr& robot_model = robot_model_loader.getModel();
+    robot_model_loader::RobotModelLoader robot_model_loader(node_, ROBOT_DESCRIPTION);
+    const robot_model::RobotModelPtr& robot_model = robot_model_loader.getModel();
     moveit::core::RobotState robot_state(robot_model);
     typedef std::map<std::string, double> JointPoseMap;
     JointPoseMap joints;
 
     robot_state.setToDefaultValues();  // initialize all joint values (just in case...)
-    for (int i = 0, end = param.size(); i != end; ++i)
+
+    rcl_interfaces::msg::ListParametersResult params_result = node_->list_parameters({ param_base_name }, 2);
```

Comment:
I actually thought that this should go the other way around, meaning we should declare the parameters for each group and then check if they have been specified. Eventually, we want to only support declared parameters if possible. Since we don't really have a best-practice for this yet, could you add a TODO to fix this in a follow-up?

Reply:
add todo 8af3088